### DISCUSSION TITLE
Samplers and schedulers

### DISCUSSION THREAD

**Discussion by dezvin** [lllyasviel/Fooocus#3201] [Q&A] [answered]
Hello and have a good day! I am playing with models from CivitAi to create my own presets and I have a few questions.

1. Could you explain please the difference between samplers that end with sde_gpu and sde? Which one should I use?
2. If I see on CivitAi that the EulerA or other sampler was used (and i dont see mention about karras, AYS, etc), which scheduler should I use? Normal? Simple? Karras?
3. Which rule of selecting a sampler and scheduler should I follow for optimal results if it is not specified in the model description?

Thank you in advance!

**Comment by mashb1t** [comment, marked as answer]
1. GPU samplers ~~are not deterministic~~ do run on GPU, which can introduce differences in denoising between GPU vendors and cards => different results, but close enough, CPU samplers are 100% the same on every machine, but are also slow. Also see https://www.reddit.com/r/comfyui/comments/1c55luw/what_is_the_point_of_nongpu_sde_samplers/
2. Karras is mostly the default when nothing has been specified.
3. https://stable-diffusion-art.com/samplers/

**Reply by dezvin** [reply]
Thank you!
